nixos/modules/config/networking.nix

```diff
@@ -309,8 +309,11 @@ in
           ln -s /run/systemd/resolve/resolv.conf /run/resolvconf/interfaces/systemd
         ''}
 
-        # Make sure resolv.conf is up to date if not managed manually or by systemd
-        ${optionalString (!config.environment.etc?"resolv.conf") ''
+        # Make sure resolv.conf is up to date if not managed manually, by systemd or
+        # by NetworkManager
+        ${optionalString (!config.environment.etc?"resolv.conf" &&
+                          cfg.networkmanager.enable ->
+                            cfg.networkmanager.rc-manager == "resolvconf") ''
```

Comment:
I'm very confused by this condition here, can you explain this?

Reply:
My bad, I had the precedence messed up there. It's supposed to check that there's no "resolv.conf"-attribute in etc and that, if networkmanager is enabled, rc-manager is set to "resolvconf".

Reply:
I'm also curious, i don't know much about network manager, but after reading the doc it seems that network manager is supposed to run itself the resolvconf program, so I'm not sure why we run here resolvconf. I would have written instead just `config.environment.etc?"resolv.conf" && !cfg.networkmanager.enable`. Did I missed something? Otherwise, thanks for this PR, it solves the problem of no internet after each rebuild!

Reply:
NetworkManager runs resolvconf only if configured to do so via `rc-manager = "resolvconf"`, which is why the condition is needed - it makes sure that if NetworkManager is enabled, resolvconf is only run here if NetworkManager is configured to use resolvconf.